tests/in/bounds-check-restrict.wgsl

```diff
@@ -0,0 +1,63 @@
+// Tests for `naga::back::BoundsCheckPolicy::Restrict`.
+
+[[block]]
+struct Globals {
+    a: array<f32, 10>;
+    v: vec4<f32>;
+    m: mat3x4<f32>;
```

Comment:
do we have a test ensuring that the policy is omitted for cases where we already know the index is in bounds?

Reply:
Good idea. Added.